src/store/hooks/useNextStep.ts

```diff
@@ -88,24 +111,89 @@ export function useNextStep() {
       storeDispatch(setIframeAnswers([]));
     }
 
-    navigate(`${computedTo}${window.location.search}`);
-  }, [
-    setIframeAnswers,
-    storageEngine,
-    storeDispatch,
-    storedAnswer,
-    trialValidation,
-    navigate,
-    startTime,
-    currentStep,
-    saveTrialAnswer,
-    computedTo,
-    windowEvents,
-    identifier,
-  ]);
+    let nextStep = currentStep + 1;
+
+    // Traverse through the sequence to find the block the current component is in
+    const blocksForStep = findBlockForStep(sequence, currentStep);
+
+    // If the current component is in a block that has a skip block (or is nested in a block that has a skip block), we need to check if the skip block should be triggered
+    const hasSkipBlock = blocksForStep !== null && (blocksForStep.some((block) => Object.hasOwn(block.currentBlock, 'skip') && block.currentBlock.skip !== undefined));
+
+    // Get the answers with the new answer added, since above is dispatching and async, but we need it synchronously
+    const answersWithNewAnswer = {
+      ...answers,
+      [identifier]: {
+        answer, startTime, endTime, provenanceGraph, windowEvents: currentWindowEvents,
+      },
+    };
+
+    // Check if the skip block should be triggered
+    if (hasSkipBlock && answersWithNewAnswer[identifier]) {
```

Comment:
Is `answersWithNewAnswer[identifier]` necessary here? It looks like above, this must always be defined.

Reply:
Fixed in 62e610f